src/agent/agent.c

```diff
@@ -1805,47 +1805,51 @@ static bool ensure_orch_address(Agent *agent) {
         if (agent->orch_addr != NULL) {
                 return true;
         }
-
-
         if (agent->host == NULL) {
                 hirte_log_errorf("No manager host specified for agent '%s'", agent->name);
                 return false;
         }
 
+        char *ip_address = agent->host;
+        bool host_is_ipv4 = isIPv4Addr(ip_address);
+        bool host_is_ipv6 = isIPv4Addr(ip_address);
 
-        if (!isIPv4Addr(agent->host) && !isIPv6Addr(agent->host)) {
-                char *ip_address = malloc0_array(0, sizeof(char), INET6_ADDRSTRLEN);
-                int r = get_address(agent->host, &ip_address);
-                if (r != 0) {
-                        hirte_log_errorf("Failed to get IP address from: '%s'", agent->host);
+        _cleanup_free_ char *resolved_ip_address = NULL;
```

Comment:
Probably move this one to line 1827?

Reply:
The purpose of `resolved_ip_address` is to store the resolved ip address string and free the allocated memory - if required (entering the if). Moving it into the if would limit its lifetime to that block so that `ip_address` would then point to a freed char*.